src/types.jl

```diff
@@ -152,13 +152,54 @@ end
     du
 end
 
+# hacking DiffCache to handling with BigFloat case
+@concrete struct BigFloatDiffCache{T <: AbstractArray, S <: AbstractArray}
+    du::T
+    dual_du::S
+    any_du::Vector{Any}
+end
+
+function BigFloatDiffCache(u::AbstractArray{T}, siz, chunk_sizes) where {T}
+    x = adapt(ArrayInterface.parameterless_type(u),
+        zeros(T, prod(chunk_sizes .+ 1) * prod(siz)))
+    xany = Any[]
+    BigFloatDiffCache(u, x, xany)
+end
+function BigFloatDiffCache(u::AbstractArray, N::Int = ForwardDiff.pickchunksize(length(u));
+    levels::Int = 1)
+    BigFloatDiffCache(u, size(u), N * ones(Int, levels))
+end
+BigFloatDiffCache(u::AbstractArray, N::AbstractArray{<:Int}) = BigFloatDiffCache(u, size(u), N)
+function BigFloatDiffCache(u::AbstractArray, ::Type{Val{N}}; levels::Int = 1) where {N}
+    BigFloatDiffCache(u, N; levels)
+end
+BigFloatDiffCache(u::AbstractArray, ::Val{N}; levels::Int = 1) where {N} = BigFloatDiffCache(u, N; levels)
+
+function get_tmp(dc::BigFloatDiffCache, u::T) where {T <: ForwardDiff.Dual}
+    nelem = length(dc.du)
+    PreallocationTools._restructure(dc.du, view(T.(dc.dual_du), 1:nelem))
+end
+function get_tmp(dc::BigFloatDiffCache, u::AbstractArray{T}) where {T <: ForwardDiff.Dual}
+    nelem = length(dc.du)
+    PreallocationTools._restructure(dc.du, view(T.(dc.dual_du), 1:nelem))
+end
```

Comment:
Don't you just need to specialize that if it's not a bitstype then you don't reinterpret? Just fix PreallocationTools.jl here?

Reply:
I can make the PR

Reply:
> Don't you just need to specialize that if it's not a bitstype then you don't reinterpret? Just fix PreallocationTools.jl here?

Yeah, I will just specialize here.